lib/routes/smartlink/index.ts

```diff
@@ -0,0 +1,75 @@
+import { Route } from '@/types';
+import ofetch from '@/utils/ofetch';
+
+function parseTitle(smartlinkUrl: string): string {
+    try {
+        const url = new URL(smartlinkUrl);
+        const pathSegments = url.pathname.split('/').filter((segment) => segment.length > 0);
+
+        // Find the segment after the date (YYYY-MM-DD pattern)
+        const dateRegex = /^\d{4}-\d{2}-\d{2}$/;
+        const dateIndex = pathSegments.findIndex((segment) => dateRegex.test(segment));
+
+        // Use the segment after the date if found, otherwise use the last path segment
+        const titleSlug = dateIndex !== -1 && dateIndex < pathSegments.length - 1 ? pathSegments[dateIndex + 1] : pathSegments.at(-1) || '';
+
+        // Convert hyphens to spaces and capitalize each word
+        return titleSlug
+            .split('-')
+            .map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
```

Comment:
You can use https://github.com/DIYgod/RSSHub/blob/d5575cf89742a9227ee25cf7bff1239c52c87748/lib/utils/common-utils.ts#L6